crossdock/crossdock_test.go

```diff
@@ -123,12 +113,45 @@ func TestCrossdock(t *testing.T) {
 	}
 }
 
+func wait(t *testing.T) {
+	attempts := 0
+	for {
+		if attempts > 9 {
+			t.Fatalf("could not talk to client in %d attempts", attempts)
+		}
+
+		attempts++
+
+		ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+		defer cancel()
```

Comment:
My understanding is that the timeouts will fire before the cancel, but I should defer a cancel() call to be safe. If that isn't the case perhaps I can just `ctx, _ := ...` instead.